### DISCUSSION TITLE
DocBin attributes- which are really necessary?

### DISCUSSION THREAD

**Discussion by violenil** [explosion/spaCy#8618] [Help: Coding & Implementations] [answered]
Hi!

I am trying to train a transformed-based model with a large amount of training data. Unfortunately, I run into memory issues when trying to preprocess the data into a training `.spacy` file. I think this happens when calling `DocBin().to_bytes()`, which 
results in a huge memory spike when lots of `Doc` objects are present in it.

Looking at the `attrs` of the `DocBin` object, these seem to be by default `("ORTH", "NORM", "TAG", "HEAD", "DEP", "ENT_IOB", "ENT_TYPE", "ENT_KB_ID", "ENT_ID", "LEMMA", "MORPH", "POS", "SENT_START")`. I suspect the memory usage can be reduced a lot by providing only the required attributes while initializing the `DocBin`.

Since I am only interested in predicting a sequence of tags/labels for each doc, is there a sure way to know which `attrs` are actually not needed in my model? I would suspect that leaving these out will result in a substantially smaller and more manageable DocBin.

Thanks so much for any help in advance!

**Comment by polm** [comment, marked as answer]
>  Unfortunately, I run into memory issues when trying to preprocess the data into a training .spacy file. 

You can split up your input files to create multiple `spacy` files and put them in the same directory. When training you can point at the directory and it will be loaded just as a single file would be. I would recommend you do that.

> is there a sure way to know which attrs are actually not needed in my model? 

There's no function that lists that for you, but I think the total list of used attributes can be found in your tok2vec spec in your config.

**Reply by violenil** [reply]
I've implemented your suggestion, with all training `spacy` files in one directory. That did the trick. Thanks a lot!
